File: 500.keyboard-row.py
#
# @lc app=leetcode id=500 lang=python3
#
# [500] Keyboard Row
#

# @lc code=start
class Solution:
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        rowdict = {}
        for c in "qwertyuiopQWERTYUIOP":
            rowdict[c] = 1
        for c in "asdfghjklASDFGHJKL":
            rowdict[c] = 2
        for c in "zxcvbnmZXCVBNM":
            rowdict[c] = 3
        res = []
        for word in words:
            if len(set(rowdict[c] for c in word)) == 1:
                res.append(word)
        return res
# @lc code=end

File: C#/500.keyboard-row.cs
/*
 * @lc app=leetcode id=500 lang=csharp
 *
 * [500] Keyboard Row
 */

// @lc code=start
using System.Collections.Generic;
public class Solution {
public string[] FindWords(string[] words) 
{
	string[] rows = new string[]{ "QWERTYUIOP", "ASDFGHJKL", "ZXCVBNM" };
	int[] keyMasks = new int[rows.Length];

	for(int i = 0; i < rows.Length; i++)
	{
		int mask = 0;
		foreach(char c in rows[i])
		{
			mask |= (1 << (c - 'A'));
		}

		keyMasks[i] = mask;
	}

	List<string> result = new List<string>();
	foreach(var word in words)
	{
		bool isValid = false;            
		int mask = 0;

		foreach(char c in word)
		{                
			mask = mask | (1 << (char.ToUpper(c) - 'A'));
		}

		for(int i = 0; i < keyMasks.Length; i++)
		{
			if((keyMasks[i] & mask) == mask)
			{
				isValid = true;
				break;
			}
		}

		if(isValid)
		{
			result.Add(word);
		}
	}

	return result.ToArray();
}
}
// @lc code=end

